modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/executor/NotificationTemplateObjectActionExecutorImpl.java

```diff
@@ -63,32 +59,53 @@ public void execute(
 				_dtoConverterRegistry, objectDefinition, payloadJSONObject,
 				_systemObjectDefinitionManagerRegistry));
 
-		notificationType.sendNotification(
-			new NotificationContextBuilder(
-			).className(
-				objectDefinition.getClassName()
-			).classPK(
-				GetterUtil.getLong(termValues.get("id"))
-			).externalReferenceCode(
-				GetterUtil.getString(termValues.get("externalReferenceCode"))
-			).groupId(
-				GetterUtil.getLong(termValues.get("groupId"))
-			).notificationTemplate(
-				notificationTemplate
-			).termValues(
-				termValues
-			).userId(
-				userId
-			).portletId(
-				objectDefinition.isUnmodifiableSystemObject() ?
-					StringPool.BLANK : objectDefinition.getPortletId()
-			).preferredLanguageId(
-				payloadJSONObject.getString("preferredLanguageId")
-			).usePreferredLanguageForGuests(
-				GetterUtil.getBoolean(
-					parametersUnicodeProperties.get(
-						"usePreferredLanguageForGuests"))
-			).build());
+		UserNotificationDelivery userNotificationDelivery =
+			_userNotificationDeliveryLocalService.getUserNotificationDelivery(
+				userId, objectDefinition.getPortletId(),
+				_classNameLocalService.getClassNameId(
+					objectDefinition.getClassName()),
+				UserNotificationDefinition.NOTIFICATION_TYPE_UPDATE_ENTRY,
+				UserNotificationDeliveryConstants.TYPE_WEBSITE, true);
+
+		if (userNotificationDelivery.isDeliver()) {
```

Comment:
@mariuo I found a method that is most commonly used for this purpose:

````java
UserNotificationManagerUtil.isDeliver(
		userId, objectDefinition.getPortletId(), 
                _classNameLocalService.getClassNameId(objectDefinition.getClassName()),
		UserNotificationDefinition.NOTIFICATION_TYPE_UPDATE_ENTRY,
		UserNotificationDeliveryConstants.TYPE_WEBSITE)
````

please use it to simplify your changes.

Thanks!